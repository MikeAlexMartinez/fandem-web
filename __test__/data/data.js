const casual = require("casual");

casual.seed(777);

const fakeUser = () => ({
  __typename: "User",
  id: casual.uuid,
  displayName: casual.word,
  name: casual.full_name,
  phoneNumber: casual.phone,
  email: casual.email,
  userRoles: [
    {
      __typename: "UserRole",
      name: "USER"
    }
  ],
  followers: {
    __typename: "UserFollower",
    followers: []
  },
  country: null,
  profilePicture: [],
  influencers: {
    __typename: "UserInfluencer",
    influencers: []
  },
  countryCode: null,
  isPrivate: false,
  subscriptions: [
    {
      __typename: "Subscription",
      name: "FREE"
    }
  ],
  emailValidated: false,
  favoriteTeam: null
});

const fakeFavoriteTeam = () => ({
  name: "Spurs",
  shortName: "TOT"
});

export { fakeUser, fakeFavoriteTeam };
