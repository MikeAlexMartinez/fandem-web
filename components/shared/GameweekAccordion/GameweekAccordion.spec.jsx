import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import GameweekAccordionComponent from './GameweekAccordion';

import { fakeGameData } from '../../../__test__/data/data';

const gameData = fakeGameData('testSeasonId');
const gameweeks = gameData[0].events;

describe('<GameweekAccordion />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(
      <GameweekAccordionComponent
        gameweeks={gameweeks}
        openGameweek={1}
        toggleGameweek={() => {}}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
