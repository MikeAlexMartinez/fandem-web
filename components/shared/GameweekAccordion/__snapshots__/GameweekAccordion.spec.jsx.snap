// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameweekAccordion /> renders and matches the snapshot 1`] = `
<GameweekAccordion
  classes={
    Object {
      "heading": "GameweekAccordion-heading-4",
      "marginRight": "GameweekAccordion-marginRight-2",
      "paddingLeft": "GameweekAccordion-paddingLeft-3",
      "root": "GameweekAccordion-root-1",
      "secondaryHeading": "GameweekAccordion-secondaryHeading-5",
    }
  }
  gameweeks={
    Array [
      Object {
        "__typename": "Gameweek",
        "deadlineTime": "2019-08-09T18:00:00.000Z",
        "fixtures": Array [
          Object {
            "__typename": "Fixture",
            "finished": false,
            "fixtureId": 1,
            "fplCode": 1059702,
            "id": "cjycw2ups0i3s0787mh7ya6x6",
            "kickoffTime": "2019-08-09T19:00:00.000Z",
            "minutes": 0,
            "teamA": Object {
              "__typename": "AwayTeamFixture",
              "awayTeam": Object {
                "__typename": "Team",
                "name": "Norwich",
              },
            },
            "teamAScore": null,
            "teamH": Object {
              "__typename": "HomeTeamFixture",
              "homeTeam": Object {
                "__typename": "Team",
                "name": "Liverpool",
              },
            },
            "teamHScore": null,
          },
          Object {
            "__typename": "Fixture",
            "finished": false,
            "fixtureId": 8,
            "fplCode": 1059709,
            "id": "cjycw2upt0i3t0787b32wik42",
            "kickoffTime": "2019-08-10T11:30:00.000Z",
            "minutes": 0,
            "teamA": Object {
              "__typename": "AwayTeamFixture",
              "awayTeam": Object {
                "__typename": "Team",
                "name": "Man City",
              },
            },
            "teamAScore": null,
            "teamH": Object {
              "__typename": "HomeTeamFixture",
              "homeTeam": Object {
                "__typename": "Team",
                "name": "West Ham",
              },
            },
            "teamHScore": null,
          },
          Object {
            "__typename": "Fixture",
            "finished": false,
            "fixtureId": 2,
            "fplCode": 1059703,
            "id": "cjycw2upu0i3u0787n9fpb3nx",
            "kickoffTime": "2019-08-10T14:00:00.000Z",
            "minutes": 0,
            "teamA": Object {
              "__typename": "AwayTeamFixture",
              "awayTeam": Object {
                "__typename": "Team",
                "name": "Sheffield Utd",
              },
            },
            "teamAScore": null,
            "teamH": Object {
              "__typename": "HomeTeamFixture",
              "homeTeam": Object {
                "__typename": "Team",
                "name": "Bournemouth",
              },
            },
            "teamHScore": null,
          },
          Object {
            "__typename": "Fixture",
            "finished": false,
            "fixtureId": 3,
            "fplCode": 1059704,
            "id": "cjycw2upu0i3v0787r7k4u2jo",
            "kickoffTime": "2019-08-10T14:00:00.000Z",
            "minutes": 0,
            "teamA": Object {
              "__typename": "AwayTeamFixture",
              "awayTeam": Object {
                "__typename": "Team",
                "name": "Southampton",
              },
            },
            "teamAScore": null,
            "teamH": Object {
              "__typename": "HomeTeamFixture",
              "homeTeam": Object {
                "__typename": "Team",
                "name": "Burnley",
              },
            },
            "teamHScore": null,
          },
        ],
        "fplEventId": 1,
        "id": "cjycw2b3w0fzg0787mmcwrghl",
        "isCurrent": false,
        "isNext": true,
        "isPrevious": false,
        "name": "Gameweek 1",
      },
    ]
  }
  openGameweek={1}
  toggleGameweek={[Function]}
/>
`;
